ToolChain/Assembler/Parser/test_Parser.py

```diff
@@ -37,1824 +37,1272 @@
-        self.assertEqual(buildInstruction.instructionCode, 0b10010000)
-        self.assertEqual(buildInstruction.sourceRegister, 0)
-        self.assertEqual(buildInstruction.sourceImmediate, None)
-        self.assertEqual(buildInstruction.destinationRegister, 1)
-        self.assertEqual(buildInstruction.destinationImmediate, None)
-        self.assertEqual(buildInstruction.width, None)
-        self.assertEqual(buildInstruction.flags, None)
-        self.assertEqual(buildInstruction.operationMnemonic, "XOR")
-        self.assertEqual(buildInstruction.instructionLength, 2)
-
-        buildInstruction, lineInstruction = self.parser._parseCodeLine(["XOR", "#0xFF", "$B"])
-        self.assertEqual(buildInstruction.instructionCode, 0b01100011)
-        self.assertEqual(buildInstruction.sourceRegister, None)
-        self.assertEqual(buildInstruction.sourceImmediate, 0xFF)
-        self.assertEqual(buildInstruction.destinationRegister, 1)
-        self.assertEqual(buildInstruction.destinationImmediate, None)
-        self.assertEqual(buildInstruction.width, None)
-        self.assertEqual(buildInstruction.flags, None)
-        self.assertEqual(buildInstruction.operationMnemonic, "XOR")
-        self.assertEqual(buildInstruction.instructionLength, 6)
-
-    def test_parseCodeLineVarious(self):
+
+        self.assertRaises(ValueError, self.parser._findInstructionCode, "InsWidthImmImm", ["error"])
+        self.assertRaises(ValueError, self.parser._findInstructionCode, "Ins", ["MEMW"])
+
+    def test_findInstructionCodeInsWidthImmReg(self):
         """
-        Test the method:
-        _parseCodeLine(self, line: list=[])
-        """
-
-        # XOR with comment
-        buildInstruction, lineInstruction = self.parser._parseCodeLine(["XOR", "#0xFF", "$B", ";test"])
-        self.assertEqual(buildInstruction.instructionCode, 0b01100011)
-        self.assertEqual(buildInstruction.sourceRegister, None)
-        self.assertEqual(buildInstruction.sourceImmediate, 0xFF)
-        self.assertEqual(buildInstruction.destinationRegister, 1)
-        self.assertEqual(buildInstruction.destinationImmediate, None)
-        self.assertEqual(buildInstruction.width, None)
-        self.assertEqual(buildInstruction.flags, None)
-        self.assertEqual(buildInstruction.operationMnemonic, "XOR")
-        self.assertEqual(buildInstruction.instructionLength, 6)
-
-        # Comment alone
-        buildInstruction, lineInstruction = self.parser._parseCodeLine([";test"])
-        self.assertIsNone(buildInstruction)
-        self.assertIsNone(lineInstruction)
-
-        # Memory location alone
-        buildInstruction, lineInstruction = self.parser._parseCodeLine(["test:"])
-        self.assertIsNone(lineInstruction)
-        self.assertEqual(buildInstruction.instructionCode, "TEST")
-        self.assertEqual(buildInstruction.sourceRegister, None)
-        self.assertEqual(buildInstruction.sourceImmediate, None)
-        self.assertEqual(buildInstruc
```

Comment:
few issues here:
- What is assembler fails silently? then the `Loneball.o` will not exist, be invalid, or use the results of a previous run
- If assembler crashes, the entire testing session crashes.
- Opening files is not verified.
- On finish, resulting file is not deleted.